src/app/info/ClientSideHelpers.tsx

```diff
@@ -1,18 +1,34 @@
 "use client";
 
-import { RoleUpdateContext } from "@/app/roles";
-import { useContext } from "react";
+import { DbWikiRow } from "@/databaseUtils";
+import { useState } from "react";
+import DefaultWikiItemView from "@/app/info/DefaultWikiItemView";
+import EditWikiItemView from "@/app/info/EditWikiItemView";
 
-export const buttonAlert = (): void => {
-    alert("Wiki item has been updated!");
+export const enterEditMode = (setIsInEditMode: (isInEditMode: boolean) => void): void => {
+    setIsInEditMode(true);
 };
 
-interface RoleProps {
-    children?: React.ReactNode;
+interface EditProps {
+    isInEditMode: boolean;
+    row?: DbWikiRow;
 }
 
-export const AdminManagerDependent: React.FC<RoleProps> = (props) => {
-    const { role } = useContext(RoleUpdateContext);
-
-    return <>{(role === "admin" || role === "manager") && props.children}</>;
+export const EditModeDependentItem: React.FC<EditProps> = ({ isInEditMode: isInEditMode, row }) => {
+    const [rowRef, setRowRef] = useState<DbWikiRow | undefined>(row);
+    const [isInEditModeRef, setIsInEditModeRef] = useState(isInEditMode);
+    return (
+        <>
+            {rowRef &&
+                (!isInEditModeRef ? (
+                    <DefaultWikiItemView rowRef={rowRef} setIsInEditModeRef={setIsInEditModeRef} />
+                ) : (
+                    <EditWikiItemView
+                        rowRef={rowRef}
+                        setRowRef={setRowRef}
+                        setIsInEditModeRef={setIsInEditModeRef}
+                    />
```

Comment:
Can we also wrap this in an admin/manager requirement? It can still be accessed by staff currently with a little bit of "behind the scenes work", even if the database calls get blocked